integration/fargate_test.go

```diff
@@ -0,0 +1,173 @@
+// +build integration
+
+package integration_test
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/dlespiau/kube-test-harness"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	. "github.com/weaveworks/eksctl/integration/runner"
+	"github.com/weaveworks/eksctl/pkg/utils/names"
+)
+
+var _ = Describe("(Integration) Fargate", func() {
+
+	// Fargate is not supported in us-west-2 yet
+	const region = "ap-northeast-1"
+
+	deleteCluster := func(clusterName string) {
+		cmd := eksctlDeleteCmd.WithArgs(
+			"cluster", clusterName,
+			"--verbose", "4",
+			"--region", region,
+		)
+		Expect(cmd).To(RunSuccessfully())
+	}
+
+	type fargateTest struct {
+		clusterName string
+		kubeTest    *harness.Test
+	}
+
+	setup := func(ft *fargateTest, createArgs ...string) {
+		ft.clusterName = "fargate-" + names.ForCluster("", "")
+		args := []string{
+			"cluster",
+			"--name", ft.clusterName,
+			"--verbose", "4",
+			"--region", region,
+			"--kubeconfig", kubeconfigPath,
+		}
+
+		args = append(args, createArgs...)
+		cmd := eksctlCreateCmd.WithArgs(args...)
+		Expect(cmd).To(RunSuccessfully())
+
+		var err error
+		ft.kubeTest, err = newKubeTest()
+		Expect(err).ToNot(HaveOccurred())
+	}
+
+	assertFargateDefaultProfile := func(clusterName string, kubeTest *harness.Test) {
+		By("having a default Fargate profile")
+		cmd := eksctlGetCmd.WithArgs(
+			"fargateprofile",
+			"--cluster", clusterName,
+			"--verbose", "4",
+			"--region", region,
+		)
+		Expect(cmd).To(RunSuccessfullyWithOutputString(ContainSubstring("fp-default")))
+
+		By("scheduling pods matching the default profile onto Fargate")
+		d := kubeTest.CreateDeploymentFromFile("default", "podinfo.yaml")
+		kubeTest.WaitForDeploymentReady(d, 5*time.Minute)
+
+		pods := kubeTest.ListPodsFromDeployment(d)
+		Expect(len(pods.Items)).To(Equal(2))
+		for _, pod := range pods.Items {
+			Expect(pod.Spec.NodeName).To(HavePrefix("fargate-"))
+		}
+		cmd = eksctlDeleteCmd.WithArgs(
+			"fargateprofile",
+			"--cluster", clusterName,
+			"--name", "fp-default",
+			"--region", region,
+			"--wait",
+			"--verbose", "4",
+		)
+		Expect(cmd).To(RunSuccessfully())
+	}
+
+	assertFargateNewProfile := func(clusterName string, kubeTest *harness.Test) {
```

Comment:
since it's not only asserting what do you think about renaming this to `testCreateFargateProfile`?

Reply:
Sounds good. I'll rename the other to `testDefaultFargateProfile`